src/bthome_ble/parser.py

```diff
@@ -555,17 +560,30 @@ def _decrypt_bthome(
         else:
             uuid = b"\xd2\xfc" + bytes([adv_info])
         encrypted_payload = data[:-8]
-        count_id = data[-8:-4]
+        last_encryption_counter = self.encryption_counter
```

Comment:
Is it possible to only update self.encryption_counter when the new value is greater than the current value? That'll avoid having the value reset after a replay.

Reply:
Done, it now only updates if the value is equal or larger.